### PR TITLE
feature: add wechat pay signature plugin

### LINKED ISSUE #1900: request help: signature plugin
Hope apisix will add signature verify plugin

### PR DESCRIPTION
**Pull request by UnderTreeTech**
### What this PR does / why we need it:
fix #1900. It's pity that i'm not familiar with nginx test case.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/plugins/signature.lua [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+function _M.check_schema(conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+
+    if not conf.algorithm then
+        conf.algorithm = "md5"
```
Comment (membphis) [inline apisix/plugins/signature.lua:82]: we can use `default` in schema

File: apisix/plugins/signature.lua [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    end
+
+    if not conf.timeout then
+        conf.timeout = 10
```
Comment (membphis) [inline apisix/plugins/signature.lua:86]: ditto

File: apisix/plugins/signature.lua [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    end
+
+    if conf.policy == "redis" then
+        if not conf.redis_host then
+            return false, "missing valid redis option host"
+        end
+
+        conf.redis_port = conf.redis_port or 6379
```
Comment (membphis) [inline apisix/plugins/signature.lua:94]: ditto

File: doc/plugins/signature.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+Here's an example, enable the `signature` plugin on the specified route:
+
+```shell
+curl -i http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "uri": "/api/verify",
+    "plugins": {
+        "signature": {
+        	"appkey":"your_appkey",
```
Comment (membphis) [inline doc/plugins/signature.md:70]: bad indentation

File: doc/zh-cn/plugins/signature.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+这就表示 `signature` 插件生效了。
+
+#### 移除插件
+
+当你想去掉 `signature` 插件的时候，很简单，在插件的配置中把对应的 json 配置删除即可，无须重启服务，即刻生效：
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "methods": ["GET"],
```
Comment (membphis) [inline doc/zh-cn/plugins/signature.md:132]: this is useless, we can remove this line

**Comment by moonming** [comment]
@UnderTreeTech any update?

**Pull request closed without merging**